introspector/src/core/telemetry_subscription.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+// This file is part of polkadot-introspector.
+//
+// polkadot-introspector is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// polkadot-introspector is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with polkadot-introspector.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+use super::TelemetryFeed;
+use color_eyre::Report;
+use futures::SinkExt;
+use futures_util::StreamExt;
+use log::{info, warn};
+use priority_channel::{SendError, Sender};
+use subxt::utils::H256;
+use tokio::{net::TcpStream, sync::broadcast::Sender as BroadcastSender};
+use tokio_tungstenite::{connect_async, tungstenite::Message, MaybeTlsStream, WebSocketStream};
+use url::Url;
+
+#[derive(Debug)]
+pub enum TelemetryEvent {
+	/// New message
+	NewMessage(TelemetryFeed),
+}
+
+pub struct TelemetrySubscription {
+	/// One sender per consumer per URL.
+	consumers: Vec<Sender<TelemetryEvent>>,
+}
+
+impl TelemetrySubscription {
+	pub fn new(consumers: Vec<Sender<TelemetryEvent>>) -> Self {
+		Self { consumers }
+	}
+
+	// Sets up per websocket tasks to handle updates and reconnects on errors.
```

Comment:
This comment seems to not reflect what the function does. This is actually the task.